ui/app/contexts/metametrics.js

```diff
@@ -55,45 +52,65 @@ export function MetaMetricsProvider ({ children }) {
     return unlisten
   }, [history])
 
+  /**
+   * track a metametrics event
+   *
+   * @param {import('../../../shared/modules/metametrics').MetaMetricsEventPayload} - payload for event
+   * @returns undefined
+   */
+  const trackEvent = useMemo(() => {
+    const referrer = confirmTransactionOrigin ? { url: confirmTransactionOrigin } : undefined
+    const page = {
+      path: currentPath,
+    }
+    return getTrackMetaMetricsEvent(global.platform.getVersion(), () => ({
+      context: {
+        referrer,
+        page,
+      },
+      environmentType,
+      network,
+      participateInMetaMetrics,
+      metaMetricsId,
+      metaMetricsSendCount,
+    }))
+  }, [network, environmentType, participateInMetaMetrics, currentPath, confirmTransactionOrigin, metaMetricsId, metaMetricsSendCount])
+
   const metricsEvent = useCallback((config = {}, overrides = {}) => {
     const { eventOpts = {} } = config
-    const { name = '' } = eventOpts
-    const { currentPath: overrideCurrentPath = '' } = overrides
-    const isSendFlow = Boolean(name.match(/^send|^confirm/u) || overrideCurrentPath.match(/send|confirm/u))
-
-    if (participateInMetaMetrics || config.isOptIn) {
-      return sendMetaMetricsEvent({
-        network,
-        environmentType,
-        activeCurrency,
-        accountType,
-        confirmTransactionOrigin,
-        metaMetricsId,
-        numberOfTokens,
-        numberOfAccounts,
-        version: global.platform.getVersion(),
-        ...config,
-        previousPath,
-        currentPath,
-        excludeMetaMetricsId: isSendFlow && !sendCountIsTrackable(metaMetricsSendCount + 1),
-        ...overrides,
-      })
+
+    const additionalContext = {}
+
+    if (overrides.currentPath) {
+      additionalContext.page = {
+        path: currentPath,
+      }
     }
 
-    return undefined
+    return trackEvent({
+      event: eventOpts.name,
+      category: eventOpts.category,
+      isOptIn: config.isOptIn,
+      excludeMetaMetricsId: eventOpts.excludeMetaMetricsId ?? overrides.excludeMetaMetricsId ?? false,
+      metaMetricsId: config.metaMetricsId,
+      matomoEvent: true,
+      properties: {
+        action: eventOpts.action,
+        number_of_tokens: numberOfTokens,
+        number_of_accounts: numberOfAccounts,
+        active_currency: activeCurrency,
+        account_type: accountType,
+        is_new_visit: config.is_new_visit,
+        ...config.customVariables,
```

Comment:
The `customVariables` here will likely be `camelCase` instead of `snake_case`. We should probably account for all possible custom variables, and convert them here. There aren't that many in total:

* `assetSelected`
* `error` (this could be consolidated with `errorMessage`)
* `errorField`
* `errorLabel`
* `errorMessage`
* `fromNetwork`
* `functionType`
* `gasChanged`
* `toNetwork`

These custom variables can be removed altogether:
* `numberOfTokens` (redundant - already added as a property)
* `numberOfAccounts` (redundant - already added as a property)
* `recipientKnown` (this is always `null`)

`origin` is also passed in as a custom variable for the transaction metrics, but this should be set to the `referrer` ideally.
* `origin` (this should be set as the `referrer`

Reply:
while i'm not against doing this, it isn't necessary. Snake case is a convention, not a requirement, and the goal here is to get the matomo events as they exist today tracking through segment -- so that we have more time to change our event schema entirely. Is your main concern here format (snake_case) or that we are explicit in what we allow to be tracked?

Reply:
Ah - I thought the intention was to change them all to `snake_case`, since you had already done that with the others.

I guess neither concern is particularly important :shrug: Formatting consistency and being explicit about possible custom variables. Mainly I wanted to make sure this wasn't an oversight.

Reply:
The goal of this pull request is to, as quickly as possible, get event tracking done through segment instead of Matomo. Lorenzo spun up a new source and destination just for this data so we can silo it from the ideal schema that he is putting together. So this data will exist separately from the "ideal" analytics data. All the other parts of the Matomo event structure (event name, category, etc) are not ideal (event names are not in Proper-case or the correct tense).

I could have been clearer about all of this in the PR description.

Reply:
Sorry, I did understand all of that - that this was a temporary solution, not intended to match any schema. I meant I wasn't sure the inconsistent casing specifically was intentional or not.

But it's fine - we can leave it as-is.

Reply:
I added a comment to show the intentional nature of not trying to conform the matomo customVars data to the existing segment schema.